### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by Phylliida** [gsgen3d/gsgen#3]
Currently my build of gs errors on Windows saying these symbols are not defined, so I need to replace them in `gs/src/include/common.h with defines like this and then it compiles ok

```
#define EPS 1e-6
#define MIN_RADIAL -85.0f
#define gs_coeff_3d 0.06349363593424097f // 1 / (2 * pi) ** (3/2)
#define gs_coeff_2d 0.15915494309189535f // 1 / (2 * pi)

#define MAX_N_FLOAT_SM 12000 // 48KB
#define MIN_RENDER_ALPHA 0.00392156862745098f // 1 / 255.0f
```

**Comment by sometimesacoder** [comment]
Note for the owners: The reason for needing #define is stated here: https://stackoverflow.com/a/56974843:
 "On windows it is using the Microsoft compiler for that. So the Microsoft compiler is correct to disallow VLA, and there is no way to avoid this AFAIK. Your code works on linux, because on linux nvcc uses the g++ host compiler, and it allows (in a non-standard-compliant way) the use of a VLA in C++ host code."

**Comment by Phylliida** [comment]
Try cloning from [my repo](https://github.com/Phylliida/gsgen/) I fixed the aux issue. Then u gotta apply the change mentioned above [to this file here](https://github.com/gsgen3d/gsgen/blob/main/gs/src/include/common.h#L83)

Now open Visual Studio command prompt and go into the gs directory. 

Download [cub](https://github.com/NVIDIA/cub) (go into releases and pick the most recent) and extract it somewhere.

In the command prompt call
SET CUB_HOME=pathuextractedcubto
SET DISTUTILS_USE_SDK=1

Now in the gs directory call

pip install -e .

If that succeeded than the scripts in the repo README should now work.

By default it'll download and install stable diffusion every time you run the code, to fix this, modify [conf/guidance/sd.yaml](https://github.com/gsgen3d/gsgen/blob/main/conf/guidance/sd.yaml#L7) to point to a local directory. The easiest way to do this is to run [example code from here](https://huggingface.co/runwayml/stable-diffusion-v1-5) then look in `C:\Users\<you>\.cache\huggingface\hub\models--runwayml--stable-diffusion-v1-5\snapshots\<some id>`

You can copy that folder to somewhere else and modify the path in sd.yaml to point to that. You might also need to change [this line](https://github.com/gsgen3d/gsgen/blob/main/conf/base.yaml#L97).

**Comment by heheyas** [comment]
Hi Phylliida,
Thank you so much for identifying and resolving these issues. However, I currently do not have access to a Windows computer, so it may take a few days before I can merge your Pull Request. Once again, thank you for your assistance! 🥰

**Comment by sometimesacoder** [comment]
These directions don't seem to work for me. To get gsgen building on my Windows 10 machine I ended up just having to add a flag to nvcc:

```
if os.name == "nt":
    nvcc_flags.append("-DWIN32_LEAN_AND_MEAN")
```

As shown here: https://github.com/sometimesacoder/gsgen-windows/commit/5179c221e26696828e60ca395dc6d3b72fdc2ea8, which I learned from here: https://github.com/facebookresearch/pytorch3d/issues/1614#issuecomment-1719514892. I don't have to download and set CUB_HOME since the compilers use the one found at `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8\include`, and I don't need the DISTUTILS_USE_SDK, I just need the one `-DWIN32_LEAN_AND_MEAN` flag. I wonder what the difference is between your environment and mine. Maybe this flag works for both of us and would be a simpler way to get things working for Windows users.

**Comment by Phylliida** [comment]
> These directions don't seem to work for me. To get gsgen building on my Windows 10 machine I ended up just having to add a flag to nvcc (after installing CUDA 11.8):
> 
> ```
> if os.name == "nt":
>     nvcc_flags.append("-DWIN32_LEAN_AND_MEAN")
> ```
> 
> As shown here: [sometimesacoder@5179c22](https://github.com/sometimesacoder/gsgen-windows/commit/5179c221e26696828e60ca395dc6d3b72fdc2ea8), which I learned from here: [facebookresearch/pytorch3d#1614 (comment)](https://github.com/facebookresearch/pytorch3d/issues/1614#issuecomment-1719514892). I don't have to download and set CUB_HOME since the compilers use the one found at `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8\include`, and I don't need the DISTUTILS_USE_SDK, I just need the one `-DWIN32_LEAN_AND_MEAN` flag. I wonder what the difference is between your environment and mine. Maybe this flag works for both of us and would be a simpler way to get things working for Windows users.
> 
> edit: to clarify, I did need to replace the `const` variables with `#define`, and renamed aux to auxiliary, it's just this CUB stuff I only needed the extra nvcc flag. Also I installed Microsoft Visual C++ build tools from https://visualstudio.microsoft.com/visual-cpp-build-tools/.

Hey very cool, that's much easier. I added that change to my repo/PR and can confirm it works for me without needing to do all that manual stuff, you can just checkout, get requirements, then pip install -e . in the gsgen/gs directory.

> PLEASEEEEEEEEEE!!!! I have slow computer >>>>I REALLY BEG YOU<<<< PLEASE RELEASE THE MODEL I WANNA DIE PLEASE WITHOUT MODEL I CANT DO ANYTHING I WANNA DIE WANNA DIE PLEASE RELEASE IT I CANT TRAIN MODEL ON THIS PC

So, the model is released? gsgen uses stable diffusion 1.5 and point-e, both of which it'll automatically download if you don't have them installed. @spacewalkingninja 
You may find [this thread](https://github.com/gsgen3d/gsgen/issues/5) useful to read
